irohad/ordering/impl/ordering_gate_impl.cpp

```diff
@@ -103,8 +109,11 @@ namespace iroha {
         std::shared_ptr<shared_model::interface::Proposal> proposal) {
       log_->info("Received new proposal, height: {}", proposal->height());
       proposal_queue_.push(std::move(proposal));
-      std::lock_guard<std::mutex> lock(proposal_mutex_);
-      net_proposals_.get_subscriber().on_next(0);
+      {
```

Comment:
Scope should not be required here. Consider the following code:
```C++
{
A a;
a.smth();
...
B b(a);
b.smth();
}
```
`B` cannot be created before `A`, so constructor will not be called when scope is opened.

Reply:
Yes, standard guarantee order of initialization by statements:
```
int a;
int b;
```
But not guarantee for the list of initialization
```
int a, b;
```

So, seems your issue is valid.